keras_nlp/src/models/resnet/resnet_backbone.py

```diff
@@ -106,6 +114,7 @@ def __init__(
         stackwise_num_strides,
         block_type,
         use_pre_activation=False,
+        use_vd_pooling=False,
```

Comment:
I don't think we want this one flag that proxies a bunch of options. This is going to be hard for maintenance, and lead to confusing configs.

I think we might want to add a new block type, `vd_block`, and two new flags here. `input_conv_filters=(64,)` and `input_conv_kernel_sizes=(7,)`. Or something like that.

Then the config for resnet vd would look like...

```python
{
    ...,
    input_conv_filters=(32, 32, 64),
    input_conv_kernel_sizes=(3, 3, 3),
    block_type="vd_block",
}
```

Then you could implement the input conv layers a a loop, and implement the new vd block, by copying the `basic_block` with some minor changes.

WDYT? @james77777778 any tweaks as well?

Not sure I'm getting the exact args right, but at a high level, avoid hard coding magic numbers that define the structure of this model behind feature flags. Instead they should be directly in the config.